subprojects/core/src/main/java/org/gradle/api/internal/file/archive/TarFileTree.java

```diff
@@ -105,6 +113,18 @@ private void visitImpl(FileVisitor visitor, InputStream inputStream) throws IOEx
         }
     }
 
+    private void checkFormat(InputStream inputStream) throws IOException {
```

Comment:
Please add a javadoc to this method to explain why it is required. One reason I remember is that unpacking of invalid TAR files silently stops with Apache Commons Compress.